test/experiment_tests.js

```diff
@@ -25,4 +24,30 @@ describe('Selector component', () => {
 			</Selector>)
 		assert.equal(selector.contains(<span>Blue</span>), true)
 	})
+
+	it ('Should return another random variant if the variant from the store is not in component', () => {
+		const initialState = {experiments: { button: 'yellow' }}
+		const mockStore = configureStore()
+		let store = mockStore(initialState)
+		shallow(<Experiment name="button" store={store}>
+				<Variant name="blue"><span>Blue</span></Variant>
+				<Variant name="red"><span>Red</span></Variant>
+				<Variant name="green"><span>Green</span></Variant>
+			</Experiment>).dive()
+		assert.equal(store.getActions().length, 1)
+		assert.equal(store.getActions()[0].type, 'RESET_EXPERIMENT_VARIANT')
```

Comment:
what does this tests ?

Reply:
If the variant is wrong, and it is not found in the Experiment children, the resetExperimentVariant will be called. And I test if it is called

Reply:
then the test name is wrong.  Based on the name of the test you should test that one of the variants is rendered